addons/sale/models/sale_order.py

```diff
@@ -1192,6 +1268,16 @@ def _get_action_add_from_catalog_extra_context(self):
     def _get_product_catalog_domain(self):
         return expression.AND([super()._get_product_catalog_domain(), [('sale_ok', '=', True)]])
 
+    def action_open_business_doc(self):
+        self.ensure_one()
+        return {
+            'name': _("Order"),
+            'type': 'ir.actions.act_window',
+            'res_model': 'sale.order',
+            'res_id': self.id,
+            'views': [(False, 'form')],
+        }
```

Comment:
I don't see where this method is used :thinking:

Reply:
This is used in `x2many_buttons` widget to open form view of duplicated record
I think accounting team added this method in py for journal because they want to open different model form [depending on record data](https://github.com/odoo/odoo/blob/master/addons/account/models/account_move.py#L4670-L4693)